### PR TITLE
Fix boolean mask indexing

### PR DESCRIPTION
**Pull request by ma-sadeghi**

### CODE REVIEW

**Comment by ma-sadeghi** [comment]
@ChrisRackauckas Could you please take a look? Also, does the GPU CI need to be manually triggered or is it part Github Actions CIs?

**Comment by ChrisRackauckas** [comment]
oh, BuildKite must not trigger on outside PRs...

**Comment by ChrisRackauckas** [comment]
I'm going rogue. I'll merge and see.

**Comment by ma-sadeghi** [comment]
@ChrisRackauckas Seems to have worked! A minor issue that I forgot to delete line 80 (which does nothing!) :cry: But, I guess should be fine (?).

https://github.com/SciML/RecursiveArrayTools.jl/blob/83fee26488c0414e91a31f042f4970d199d1d000/src/vector_of_array.jl#L76-L82

**Comment by ChrisRackauckas** [comment]
I just did a small commit removing it. No worries.

File: src/vector_of_array.jl [hunk 1]
```diff
@@ -78,7 +78,7 @@ Base.@propagate_inbounds function Base.getindex(A::AbstractVectorOfArray{T, N},
   @assert length(J) == ndims(A.u[1])+1-ndims(I)
   @assert size(I) == size(A)[1:ndims(A)-length(J)]
   vecs = vec.(A.u)
-  return Adapt.adapt(__parameterless_type(T),reshape(reduce(hcat,vecs[I]),size(A.u[1])...,length(A.u)))
+  return A[repeat([:], length(size(A)))...][I,J...]
```
Comment (YingboMa) [inline src/vector_of_array.jl:81]: Maybe `A[ntuple(_->(:), ndims(A))...][I,J...]`?

Comment (ChrisRackauckas) [reply src/vector_of_array.jl:81]: yeah the tuple would be better. Open a PR?

Comment (ma-sadeghi) [reply src/vector_of_array.jl:81]: Sure, please see #184.

**Pull request merged**